src/o365/spo/commands/app/app-add.ts

```diff
@@ -67,13 +64,21 @@ class SpoAppAddCommand extends SpoCommand {
           cmd.log(`Adding app '${fullPath}' to app catalog...`);
         }
 
+        let appCatalogUrl: string = auth.site.url;
+        if (args.options.appCatalogUrl) {
+          appCatalogUrl = args.options.appCatalogUrl as string;
+        }
+
+        if(args.options.scope === 'sitecollection') {
+          appCatalogUrl = appCatalogUrl.toLowerCase().replace('/appcatalog', '');
+        }
+
         const fileName: string = path.basename(fullPath);
         const requestOptions: any = {
-          url: `${auth.site.url}/_api/web/tenantappcatalog/Add(overwrite=${(overwrite.toString().toLowerCase())}, url='${fileName}')`,
+          url: `${appCatalogUrl}/_api/web/${scope}appcatalog/Add(overwrite=${(overwrite.toString().toLowerCase())}, url='${fileName}')`,
```

Comment:
When connected to the tenant admin site, this command fails, because here we're using the URL of the site passed in the command options, while on line 80 we're using the access token from the site used to log in.